client/v3/snapshot/v3_snapshot.go

```diff
@@ -76,26 +76,26 @@ func SaveWithVersion(ctx context.Context, lg *zap.Logger, cfg clientv3.Config, d
 	var size int64
 	size, err = io.Copy(f, resp.Snapshot)
 	if err != nil {
-		return resp.Version, err
+		return "", err
```

Comment:
Why change this? I would suggest to return the resp.Version, because it's the real value returned from the server. It's up the caller how to use it. We should only return "" when we get [an error response](https://github.com/etcd-io/etcd/blob/f3c9db9c460bba556f4922a5acc3be496eccda40/client/v3/snapshot/v3_snapshot.go#L72) from etcdserver.

This comment applies to the following couple of places.

Reply:
if the err is not nil, I think that in go practice, the caller won't use `version`.

Reply:
Overall it isn't a big deal. 

Note that `SaveWithVersion` is a public API. Previously it returns a valid version as long as the first communication with etcdserver is successful. Now the behavior changed in this PR although it isn't a big problem.

You never know how users will use a API. If I use the API, I might get the version included in an error log message anyway; and I will be a little surprised that why previously I can see a valid version but not now.

So let's only fix bugs, and try not to change any behavior unless it's necessary.